packages/desktop-client/src/components/App.tsx

```diff
@@ -59,25 +71,25 @@ function AppInner() {
         loadingText: t('Initializing the connection to the local database...'),
       }),
     );
-    await initConnection(socketName);
+    await maybeUpdate(() => initConnection(socketName));
```

Comment:
It's a great question, and I'm not sure what the best approach is here. Essentially, I previously had a different version of this where the app would _automatically_ restart after it finished downloading an update. That creates a weird UX where you could get interrupted, but it also has a benefit that on cold starts, you could get the update installed without needing a restart at all. To avoid the interruption problem, the app normally shows the pop-up (and this is done in the background automatically, with no calls to `maybeUpdate`). However, during _loading_, the idea was that if an update is made available, we stop loading and immediately restart to update. In order to implement that in the smoothest way, I have it polling right now after each step of loading.

One alternative would be to just check for an update once at the _end_ of the initial loading screen. This has the downside that the user might have to wait through a full load once, and then have the app restart, but it does result in simpler code (only one poll at the end of loading). What do you think?

Reply:
I'm for the simpler solution personally. It takes less than a second for me to get through the inital load and I think it makes more sense. I'm happy to take your lead on it though, since you wrote it.

Reply:
Let me know what you decide, I'm happy to approve this as is but if you've got a simpler solution I can wait.

Reply:
Updated to only check once at the start and once at the end. PTAL!